app/lib/request.rb

```diff
@@ -139,7 +152,7 @@ def valid_url?(url)
     end
 
     def http_client
-      HTTP.use(:auto_inflate).follow(max_hops: 3)
+      HTTP.use(:auto_inflate).use(normalize_uri: { normalizer: URI_NORMALIZER }).follow(max_hops: 3)
```

Comment:
Question, if we *know* we pre-normalize the URI in this class, couldn't this just be a no-op?

Reply:
The normalizer is also used for redirects, i.e. if the first request includes a `Location: HTTP://EXAMPLE.COM` header, we want to normalise that URL as well.

I have added a test for this.

BTW, a no-op normalizer would look like this:
```
->(x) { HTTP::URI.parse(x) }
```
i.e. a bit more complex than `->(x) { x }` as mentioned in https://github.com/mastodon/mastodon/issues/24932#issuecomment-1545419392.

Reply:
Makes sense